countComponents.js

```diff
@@ -0,0 +1,75 @@
+/*
+graph of nodes
+
+array[i] = [a, b]
+
+i: n of nodes
+o: => number of connected components
+
+Input: n = 5, edges = [[0,1],[1,2],[3,4]]
+Output: 2
+
+0 - 1 - 2
+
+3 - 4
+
+*/
+
+
+const buildGraph = (edges) => {
+    const graph = {};
+
+    for (const edge of edges) {
+        const [nodeOne, nodeTwo] = edge;
+
+        const isNodeOne = graph.hasOwnProperty(nodeOne);
+        const isNodeTwo = graph.hasOwnProperty(nodeTwo);
+
+        if (!isNodeOne) graph[nodeOne] = [];
+        if (!isNodeTwo) graph[nodeTwo] = [];
+
+        graph[nodeOne].push(nodeTwo);
+        graph[nodeTwo].push(nodeOne);
+    }
+
+    return graph;
+};
+
+const markAsVisited = (node, edges, visited) => {
```

Comment:
What are you marking as visited? You should specify that in the name. Based on your current name, I might guess that it just marks the given node as visited.